pkg/file/joiner/joiner.go

```diff
@@ -31,6 +31,18 @@ func NewSimpleJoiner(getter storage.Getter) file.Joiner {
 }
 
 func (s *simpleJoiner) Size(ctx context.Context, address swarm.Address) (dataSize int64, err error) {
+	toDecrypt := len(address.Bytes()) == (swarm.HashSize + encryption.KeyLength)
+
+	var addr []byte
+	var key encryption.Key
+	if toDecrypt {
+		addr = address.Bytes()[:swarm.HashSize]
+		key = address.Bytes()[swarm.HashSize : swarm.HashSize+encryption.KeyLength]
+	} else {
+		addr = address.Bytes()
+	}
```

Comment:
as in my previous comment, here i would define default assignments (no encryption) and then add a following `if` without `else`.

i think this makes things more readable, but it is not a blocker